asg/src/expression/array_inline.rs

```diff
@@ -118,6 +119,27 @@ impl FromAst<leo_ast::ArrayInlineExpression> for ArrayInlineExpression {
         };
 
         let mut len = 0;
+
+        if len == 0 && expected_item.is_none() {
```

Comment:
len == 0 check here is a nop

Reply:
Ok yea, I didn't like this approach either but wasn't sure what the best way to go about it was. Maybe a better approach would be to match on the result of the expr (currently on 149) and if it gives the unknown error type push that on a queue to resolve at the end of the map. Inserting the elements back in the right location was also a concern then.

Reply:
we also need to make sure, somehow, that no scoping error occurs. i.e. planning head for some kind of global state, we can't evaluate the nodes out of order really. makes me want to just not allow implicits out of order like this, because we have to violate execution order/evaluation order to infer properly.

Reply:
I know that rust allows this behavior, I can look into how they approach this situation for reference. But also agreed that it is a pain since I can't imagine they do this without some expensive checks in there. I could be wrong though.

Reply:
Okay looking into it rust just does the two passes approach. So I guess the discussion should more be, do we really need a niche feature that could be expensive? Honestly, can't say I would ever type my arrays as [1, 2, 3, ..., n-1u32] anyways unless I intentionally want to create the slowest program. Even in the case provided of [1, 2u32, ..., n-1], I don't ever encounter anything like that.